AND_2.0.0-beta3 (#24)
UnifiedPush/specifications: specifications/android.md

* message (String): a short description of the purpose of the registration that the distributor MAY show to the user.

The distributor MUST send a broadcast intent to one of the following actions when it handles this action:
* org.unifiedpush.android.connector.NEW_ENDPOINT
* org.unifiedpush.android.connector.REGISTRATION_FAILED

### org.unifiedpush.android.distributor.UNREGISTER

The connector sends this action to unregister from push messages. The intent MUST contain 1 extra:
* token (String): the token supplied by the end user application during registration

The intent SHOULD contain 1 additional extra:
* application (String): the end user application package name.

The distributor MUST send a broadcast intent to the following action when it handles this action:
* org.unifiedpush.android.connector.UNREGISTERED.

### org.unifiedpush.android.distributor.MESSAGE_ACK

Whenever the connector receives a message with the extra id it MUST reply with this action to the distributor with the String extra id received to acknowledge the message reception.

## Messaging Broadcast Receiver

> The exposed broadcast receiver of the connector library MUST handle 2 differents actions:

Reviewer: ```suggestion
The exposed broadcast receiver of the end user application MUST handle 2 differents actions:
```

* message (String): the push message sent by the application server, as a string.

If the BYTES_MESSAGE feature was not requested, it MUST send the following 2 extras:
* token (String): the token supplied by the end user application during registration
* message (String): the push message sent by the application server, as a string.

If the BYTES_MESSAGE feature was not requested, it MAY additionally send the message as a byte array:
* bytesMessage (ByteArray): the push message sent by the application server, as an array of bytes. It MUST be the raw POST data received by the rewrite proxy.

It MAY be sent with the following extra:
* id (String): to identify the message

### org.unifiedpush.android.connector.UNREGISTERED

The distributor MUST send this action to the registered application to confirm unregistration or to inform the application about unregistration. If this action was triggered by an UNREGISTER action then the distributor MUST retrieve the target application name using the token supplied. If no registration matching the token is found, then it MUST use the application received with the extra application, if it is present.

The intent MUST have the following extra:

> * token (String): the token supplied by the end user application during registration. If this action was triggered by a UNREGISTER action this is the token supplied in the extras of the triggering action.

Reviewer: ```suggestion
* token (String): the token supplied by the end user application during registration to inform about unregistration, or the token supplied in the extras of the UNREGISTER action to confirm unregistration.
```
